pkg/server/server.go

```diff
@@ -53,8 +65,17 @@ func New(opts *tlsopts.Options, lis net.Listener, interceptor grpc.UnaryServerIn
 // Identity returns the server's identity
 func (p *Server) Identity() *identity.FullIdentity { return p.identity }
 
-// Addr returns the server's listener address
-func (p *Server) Addr() net.Addr { return p.lis.Addr() }
+// PublicAddr returns the server's listener address
+func (p *Server) PublicAddr() net.Addr { return p.publicListener.Addr() }
+
+// PrivateAddr returns the server's listener address
+func (p *Server) PrivateAddr() net.Addr { return p.privateListener.Addr() }
+
+// PublicListener returns the server's public listener
+func (p *Server) PublicListener() net.Listener { return p.publicListener }
```

Comment:
I think you won't need to expose the listeners. Exposing the appropriate GRPC would be sufficient.